File: spring-webflux/src/main/java/software/darkmatter/school/blog/domain/user/domain/UserRepository.java
package software.darkmatter.school.blog.domain.user.domain;

import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface UserRepository extends ReactiveCrudRepository<User, Long> {

    Mono<User> findByIdAndDeletedAtIsNull(Long id);

    Flux<User> findAllByDeletedAtIsNull(Pageable pageable);
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/domain/user/domain/UserRepository.kt
package software.darkmatter.school.blog.domain.user.domain

import kotlinx.coroutines.flow.Flow
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.reactive.ReactiveCrudRepository

interface UserRepository : ReactiveCrudRepository<User, Long> {

    suspend fun findByIdAndDeletedAtIsNull(id: Long?): User?

    fun findAllByDeletedAtIsNull(pageable: Pageable): Flow<User>
}
